sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_requiredparameters.json

```diff
@@ -2006,7 +2007,8 @@
                     ],
                     "type": "string"
                 }
-            }
+            },
+            "required": ["type"]
```

Comment:
This one was debatable for me. When creating a device or module, authenticationMechanism as a whole can be null, and the service will just assume symmetric key based. However, if a user is building this AuthenticationMechanism object, I believe it makes sense to have them declare the type explictly

Reply:
Does the service return a Bad Request when we don't send these properties? Want to make sure we don't override any defaults they have in the service when this is not sent. This will be a breaking change in that case.

Reply:
sas case: no keys, no problem. iothub generates them
self signed case: http request still succeeds if type is self signed, but no thumbprints are given
if no authentication type is given: IotHub assume sas based, and generates keys for the user.

Looks like this field isn't required. I'll remove this required tag